maven-resolver-named-locks/src/main/java/org/eclipse/aether/named/support/FileLockNamedLock.java

```
@@ -0,0 +1,287 @@
+                }
+            }
+            finally
+            {
+                criticalRegion.unlock();
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public void unlock()
+    {
+        criticalRegion.lock();
+        try
+        {
+            Deque<Boolean> steps = threadSteps.computeIfAbsent( Thread.currentThread(), k -> new ArrayDeque<>() );
+            if ( steps.isEmpty() )
+            {
+                throw new IllegalStateException( "Wrong API usage: unlock without lock" );
+            }
+            Boolean shared = steps.pop();
+            logger.trace( "{} unlock (shared = {})", name(), shared );
+            if ( steps.isEmpty() && !anyOtherThreadHasSteps() )
+            {
+                try
+                {
+                    fileLockRef.getAndSet( null ).release();
+                }
+                catch ( IOException e )
+                {
+                    throw new UncheckedIOException( e );
+                }
+            }
+        }
+        finally
+        {
+            criticalRegion.unlock();
+        }
+    }
+
+    /**
+     * Returns {@code true} if any other than this thread using this instance has any step recorded.
+     */
+    private boolean anyOtherThreadHasSteps()
+    {
+        return threadSteps.entrySet().stream()
+                .filter( e -> !Thread.currentThread().equals( e.getKey() ) )
+                .map( Map.Entry::getValue )
+                .anyMatch( d -> !d.isEmpty() );
+    }
+
+    /**
+     * Attempts to obtain real {@link FileLock}, returns non-null value is succeeds, or {@code null} if cannot.
+     */
+    private FileLock obtainFileLock( final boolean shared )
+    {
+        FileLock result;
+        try
+        {
+            result = fileChannel.tryLock( LOCK_POSITION, LOCK_SIZE, shared );
+        }
+        catch ( OverlappingFileLockException e )
+        {
+            logger.trace( "Overlap on {}: {}", name(), e.getMessage() );
```

Don't swallow exceptions. Pass `e` as-is.

---

fixed